javascript/ql/src/Security/CWE-312/CleartextLogging.ql

```diff
@@ -0,0 +1,55 @@
+/**
+ * @name Clear text logging of sensitive information
+ * @description Sensitive information logged without encryption or hashing can expose it to an
+ *              attacker.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id js/cleartext-logging
+ * @tags security
+ *       external/cwe/cwe-312
+ *       external/cwe/cwe-315
+ *       external/cwe/cwe-359
+ */
+
+import javascript
+import semmle.javascript.security.dataflow.CleartextLogging::CleartextLogging
+
+/**
+ * Holds if `tl` is used in a browser environment.
+ */
+predicate inBrowserEnvironment(TopLevel tl) {
```

Comment:
I do not think that is a false negative.
This predicate is used to exclude alerts for logging that occurs on the user's own computer since that is innocent in practice.
Both browsers and electron apps (that use browser features) are run on the user's own computer.

Reply:
Ah, you're right.